cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -127,56 +114,22 @@ class _MigrationContext:
     """
     Context for the migration process.
     """
-    existing_source_to_target_keys: dict[  # Note: It's intended to be mutable to reflect changes during migration.
-        UsageKey, list[PublishableEntity | None]
-    ]
+    # Fields that get mutated as we migrate blocks
+    used_component_keys: set[LibraryUsageLocatorV2]
+    used_container_slugs: set[str]
```

Comment:
the only thing we need to mutate is the _set of keys and slugs that are taken_. We don't need a mutable list of source->target mappings.